.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,64 @@
+name: Build and Publish Docker Image
+
+on:
+  push:
+    branches:
+      - feat/image-pipeline #Should be set to main in future
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          driver: docker
+
+      # Build all the tags for the image
+      - name: Create tags
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ghcr.io/2pisoftware/cmfive
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+
+      # Login to GHCR
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Build the boilerplate image, the final image requires this
+      - name: Build boilerplate image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: false
+          tags: cmfive-boilerplate:latest
+          load: true # Use the built image in the next step
+
+      # Build the theme, copy it to boilerplate, tag and push
```

Comment:
Do we want a new image every time a push happens?
Or should we just do it on tags?

We will need a clean up action to remove old images over time as we do have a limit on hosting them

Reply:
Open to how we want to do the images, note that the boilerplate is not pushed to ghcr, but the one below that is and with tags etc.

I was thinking of doing on PR instead, then you'll get an image of that PR you can test with (tagged to that branch). You can then add a version tag as you wish and the docker image will also get that version tag.